lib/XTR/StatsMigrationTask.php

```diff
@@ -0,0 +1,481 @@
+<?php
+/**
+ * Copyright (C) 2019 Xibo Signage Ltd
+ *
+ * Xibo - Digital Signage - http://www.xibo.org.uk
+ *
+ * This file is part of Xibo.
+ *
+ * Xibo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Xibo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Xibo.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Xibo\XTR;
+use Xibo\Entity\User;
+use Xibo\Factory\TaskFactory;
+use Xibo\Factory\UserFactory;
+
+/**
+ * Class StatsMigrationTask
+ * @package Xibo\XTR
+ */
+class StatsMigrationTask implements TaskInterface
+{
+    use TaskTrait;
+
+    /** @var  User */
+    private $archiveOwner;
+
+    /** @var UserFactory */
+    private $userFactory;
+
+    /** @var TaskFactory */
+    private $taskFactory;
+
+    private $archiveExist;
+
+    /** @inheritdoc */
+    public function setFactories($container)
+    {
+        $this->userFactory = $container->get('userFactory');
+        $this->taskFactory = $container->get('taskFactory');
+        return $this;
+    }
+
+    /** @inheritdoc */
+    public function run()
+    {
+        $this->migrateStats();
+    }
+
+    public function migrateStats()
+    {
+        $this->runMessage = '# ' . __('Stats Migration') . PHP_EOL . PHP_EOL;
+
+        // read configOverride
+        $configOverrideFile = $this->getOption('configOverride', '');
+        if (file_exists($configOverrideFile)) {
```

Comment:
can we use an array_merge here instead, that way we wouldn't have to provide all of the settings in the override.

something like:

```
if (!empty($configOverrideFile) && file_exists($configOverrideFile)) {
    $options = array_merge( json_decode(file_get_contents($configOverrideFile), true), $options);
}
```

Reply:
its the other way around I guess

>  // read configOverride
        $configOverrideFile = $this->getOption('configOverride', '');
        if (!empty($configOverrideFile) && file_exists($configOverrideFile)) {
            $options = array_merge($options, json_decode(file_get_contents($configOverrideFile), true));
        }